### PR TITLE
Use stdlib.h for a C++ compatible NULL definition

### PR DESCRIPTION
**Pull request by mwit-a**

### CODE REVIEW

**Comment by mwit-a** [comment]
@rnestler

File: sensirion_arch_config.h [hunk 1]
```diff
@@ -38,6 +38,16 @@
  */
 #include <stdint.h>
 
+/**
+* If your platform does not provide the library stdlib.h you have to
+* define NULL yourself (see below).
+ */
+#include <stdlib.h>
```
Comment (chjaeggi) [inline sensirion_arch_config.h]: If there is no stdlib header file, this include won't work

Comment (rnestler) [reply sensirion_arch_config.h]: Right, see the comment above.

File: sensirion_arch_config.h [hunk 2]
```diff
@@ -38,6 +38,16 @@
...
+
+#ifndef NULL
+#define NULL ((void *)0)
+#endif
```
Comment (rnestler) [inline sensirion_arch_config.h]: Should we comment it out per default and mention that the user should uncomment if `stdlib.h` is not available?

File: sensirion_arch_config.h (hunk 1 above)
Comment (rnestler) [reply sensirion_arch_config.h]: But maybe change it to: 
```
/**
* If your platform does not provide the library stdlib.h you have to remove the include and
* define NULL yourself (see below).
 */
```

File: sensirion_arch_config.h (hunk 2 above)
Comment (abrauchli) [reply sensirion_arch_config.h]: I'd be ok with that

Comment (chjaeggi) [reply sensirion_arch_config.h]: Sure

File: sensirion_arch_config.h (hunk 1 above)
Comment (mwit-a) [reply sensirion_arch_config.h]: Done.

File: sensirion_arch_config.h (hunk 2 above)
Comment (mwit-a) [reply sensirion_arch_config.h]: Done.

**Comment by mwit-a** [comment]
@rnestler @abrauchli  @chjaeggi

File: sensirion_arch_config.h (hunk 2 above)
Comment (mwit-a) [reply sensirion_arch_config.h]: Done.

**Pull request merged**